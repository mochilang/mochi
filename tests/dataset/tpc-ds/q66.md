# TPC-DS Query 66

This is a placeholder implementation for TPC-DS query 66.

## SQL
```sql
SELECT 66;
```

## Expected Output
```json
66
```
