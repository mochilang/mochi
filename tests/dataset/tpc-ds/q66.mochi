let t = [{id: 1, val: 66}]
let result = from r in t select r.val |> first
json(result)

test "TPCDS Q66 placeholder" {
  expect result == 66
}
