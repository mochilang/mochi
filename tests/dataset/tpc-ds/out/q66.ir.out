func main (regs=28)
  // let web_sales = [
  Const        r0, [{"net": 30}]
  // let catalog_sales = [
  Const        r1, [{"net": 36}]
  // let result = sum(from w in web_sales select w.net) + sum(from c in catalog_sales select c.net)
  Const        r2, []
  Const        r3, "net"
  IterPrep     r4, r0
  Len          r5, r4
  Const        r7, 0
  Move         r6, r7
L1:
  LessInt      r8, r6, r5
  JumpIfFalse  r8, L0
  Index        r9, r4, r6
  Move         r10, r9
  Index        r11, r10, r3
  Append       r12, r2, r11
  Move         r2, r12
  Const        r13, 1
  AddInt       r6, r6, r13
  Jump         L1
L0:
  Sum          r14, r2
  Const        r15, []
  IterPrep     r16, r1
  Len          r17, r16
  Move         r18, r7
L3:
  LessInt      r19, r18, r17
  JumpIfFalse  r19, L2
  Index        r20, r16, r18
  Move         r21, r20
  Index        r22, r21, r3
  Append       r23, r15, r22
  Move         r15, r23
  AddInt       r18, r18, r13
  Jump         L3
L2:
  Sum          r24, r15
  Add          r25, r14, r24
  // json(result)
  JSON         r25
  // expect result == 66
  Const        r26, 66
  Equal        r27, r25, r26
  Expect       r27
  Return       r0
